File: Objective C Playground/Objective C Playground/Person.m
//
//  Person.m
//  Objective C Playground
//
//  Created by Leon Baird on 08/12/2015.
//  Copyright © 2015 Leon Baird. All rights reserved.
//

#import "Person.h"

@implementation Person

@synthesize age=privateAge;

- (instancetype)init
{
    self = [super init];
    if (self) {
        self.surname = @"Person";
        self.forename = @"New";
        self.age = 0;
    }
    return self;
}

- (NSString *)surname {
    return _surname;
}

- (void)setSurname:(NSString *)surname {
    NSString *copy = [surname copy];
    _surname = copy;
}


-(NSString *)getFullName {
    return [NSString stringWithFormat:@"%@ %@", _forename, _surname];
}

@end

File: Archiving/Archiving/Person.swift
//
//  Person.swift
//  Archiving
//
//  Created by Leon Baird on 11/12/2015.
//  Copyright © 2015 Leon Baird. All rights reserved.
//

import UIKit

class Person: NSObject, NSCoding {

    var surname:String = "New"
    var forename:String = "New"
    
    var age:Int = 0
    
    init(surname:String, forename:String, age:Int) {
        self.surname = surname
        self.forename = forename
        self.age = age
    }
    
    // MARK: - NSCoding Methods
    
    required init(coder aDecoder: NSCoder) {
        self.surname = aDecoder.decodeObjectForKey("surname") as! String
        self.forename = aDecoder.decodeObjectForKey("forename") as! String
        self.age = aDecoder.decodeIntegerForKey("age")
    }
    
    func encodeWithCoder(aCoder: NSCoder) {
        aCoder.encodeObject(surname, forKey: "surname")
        aCoder.encodeObject(forename, forKey: "forename")
        aCoder.encodeInteger(age, forKey: "age")
    }
    
}
